virtualizarr/kerchunk.py

```diff
@@ -262,9 +262,16 @@ def variable_to_kerchunk_arr_refs(var: xr.Variable, var_name: str) -> KerchunkAr
                     f"Cannot serialize loaded variable {var_name}, as it is encoded with an offset"
                 )
             if "calendar" in var.encoding:
-                raise NotImplementedError(
-                    f"Cannot serialize loaded variable {var_name}, as it is encoded with a calendar"
+                import cftime
+
+                calendar = var.attrs.get(
+                    "calendar", var.encoding.get("calendar", "standard")
                 )
+                units = var.attrs.get("units", var.encoding["units"])
+
+                data = cftime.date2num(var, calendar=calendar, units=units).ravel()
+                var = var.copy(data=data)
+                np_arr = var.to_numpy()
```

Comment:
Thinking out loud: Should this be a Zarr Codec instead?

Reply:
naive question: do you mean a new CFTime codec or just a number codec rather than numpy?

Reply:
Would that just be a refactoring in this PR? Like just separate out the logic here into a Codec class that could later be moved upstream?

Reply:
@TomNicholas I've looked at a tiny portion of upstreaming CF logic to a codec for CF `scale_factor` and `add_offset` but this case has the benefit of a pre-existing `numcodec` [FixedScaleAndOffset](https://numcodecs.readthedocs.io/en/stable/fixedscaleoffset.html).

Reply:
There is also a [`CFScaleOffsetCoder`](https://github.com/pydata/xarray/blob/b5180749d351f8b85fd39677bf137caaa90288a7/xarray/coding/variables.py#L453) in xarray too...